lib/docker-toolbelt.coffee

```diff
@@ -9,17 +9,10 @@ path = require 'path'
 randomstring = require 'randomstring'
 execAsync = Promise.promisify(require('child_process').exec)
 
-Promise.promisifyAll Docker.prototype, {
-	filter: (name) -> name == 'run'
-	multiArgs: true
-}
-Promise.promisifyAll(Docker.prototype)
-
-# Hack dockerode to promisify internal classes' prototypes
-Promise.promisifyAll(Docker({}).getImage().constructor.prototype)
-Promise.promisifyAll(Docker({}).getContainer().constructor.prototype)
-
-module.exports = Docker
+module.exports = class DockerToolbelt extends Docker
+	constructor: (opts) ->
+		opts.Promise ?= Promise
```

Comment:
It's probably worth enforcing using bluebird, as there are a bunch of places where we will return bluebird promises even if the user specifies a different type of promise - the alternative would be to respect the passed in Promise and make sure we always return promises of that type from our added methods.  Either way I think we should be consistent with the type of promise returned so that it can be relied on